manifests/localzone.pp

```diff
@@ -9,23 +9,39 @@
 # === Parameters:
 #
 # [*zone*]
-#   (required) String. Zone name
+#   (required) String. Zone name.
 #
 # [*type*]
-#   (required) Custom type Unbound::Local_zone_type
+#   (required) Custom type Unbound::Local_zone_type.
 #
 # [*config_file*]
-#   (optional) name of configuration file
+#   (optional) name of configuration file.
+#
+# [*local_data*]
+#   (optional) Array.
+#   Define local data which should be rendered.
+#   Default value: [].
+#   Example:
+#     unbound::localzone::local_data:
+#       - 'api.test.com 15 IN A 1.1.1.1'
+#       - 'backend.test.com' 15 IN A 1.1.1.1'
+#
+# [*template_name*]
+#   (optional) String.
+#   Use a custom template.
+#   Default value: 'unbound/local_zone.erb'
 #
 define unbound::localzone (
   Unbound::Local_zone_type $type,
   String $zone = $name,
   $config_file = $unbound::config_file,
+  Array[String] $local_data = [],
```

Comment:
it might be nice to create a custome type for resource records instead of just accepting any random string.  something like
```puppet
type unbound::local_data =  Struct[{
    name => String,
    type => String,
    data => String,
  }]
]
```
in yaml you would have
```yaml
unbound::local_data:
  - name: foobar
    rrtype: A
    data: 192.0.2.1
  - name: foobar
    rrtype: TXT
    data: 'spf=foobar'
```

Reply:
This seems like a good suggestion to me.  It will slightly complicate the template, but not much.

Reply:
Thanks for ideas. I adjusted PR.